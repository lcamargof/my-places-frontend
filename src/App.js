import React, { Component } from 'react';
import { Button, withStyles } from 'material-ui';
import HelpIcon from '@material-ui/icons/HelpOutline';
import MainAppBar from './components/MainAppBar';
import Map from './components/Map';
import HelpDialog from './components/HelpDialog';

const styles = theme => ({
  button: {
    margin: theme.spacing.unit * 2,
    position: 'fixed',
    bottom: 0,
    right: 0,
  }
});

class App extends Component {
  state = {
    places: [
      { id: 1, name: 'Awesome place', open: '00:00', close: '23:00', description: 'asdf', location: [-103.40932570000001, 20.6713195] },
      { id: 2, name: 'OH GOD', open: '15:00', close: '16:00', description: 'asdf', location: [-103.41932570000001, 20.6713195] },
      { id: 3, name: 'ROCKET!', open: '19:00', close: '20:00', description: 'adsdf', location: [-103.40932570000001, 20.6913195] },
    ],
    search: '',
    place: null,
    help: false,
    confirm: false,
    form: false,
    selected: null,
  };

  componentDidMount() {
    if (navigator.geolocation) {
      navigator.geolocation.getCurrentPosition(position => (
        this.setState({ location: [position.coords.longitude, position.coords.latitude] })
      ));
    }
  }

  handleFilter = (place) => {
    this.setState({ place });
  };

  handleAddPlace = (place) => {
    place.id = Math.floor(Math.random() * 100000000);

    this.setState(prevState => ({ places: [...prevState.places, place] }));
  };

  handlePlaceUpdate = (place) => {
    const index = this.state.places.findIndex(p => p.id === place.id);

    this.setState(prevState => ({
      places: [
        ...prevState.places.slice(0, index),
        place,
        ...prevState.places.slice(index + 1)
      ]
    }))
  };

  handlePlaceRemove = (place) => {
    const index = this.state.places.findIndex(p => p.id === place.id);

    this.setState(prevState => ({
      places: [
        ...prevState.places.slice(0, index),
        ...prevState.places.slice(index + 1)
      ]
    }));
  };

  toggleHelp = help => this.setState({ help });

  render() {
    const { classes } = this.props;
    const { places, place, help } = this.state;

    return (
      <div className="App">
        <MainAppBar places={this.state.places} onFilter={this.handleFilter} />
        <Map
          places={places}
          place={place}
          onAdd={this.handleAddPlace}
          onUpdate={this.handlePlaceUpdate}
          onDelete={this.handlePlaceRemove}
        />
        <Button
          variant="fab"
          color="primary"
          aria-label="help"
          className={classes.button}
          onClick={() => this.toggleHelp(true)}
        >
          <HelpIcon />
        </Button>
        <HelpDialog open={help} onClose={() => this.toggleHelp(false)} />
      </div>
    );
  }
}

export default withStyles(styles)(App);
