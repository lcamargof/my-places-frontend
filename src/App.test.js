import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as mockSocket from 'mock-socket';
import Axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { App } from './App';

jest.mock('mapbox-gl', () => ({
  Map: jest.fn(() => ({ on: jest.fn(), getCenter: jest.fn() }))
}));

// jest.mock('axios', () => ({
//   get: new Promise(resolve => resolve({ data: { success: true } })),
//   post: new Promise(resolve => resolve({ data: { success: true } })),
//   put: new Promise(resolve => resolve({ data: { success: true } })),
//   delete: new Promise(resolve => resolve({ data: { success: true } })),
// }));

jest.mock('socket.io-client', () => {
  return mockSocket.SocketIO;
});

const response = {
  data: {
    success: true,
    data: [{ id: 1 }]
  }
};

const axiosMock = new MockAdapter(Axios)
  .onGet().reply(200, response)
  .onPost().reply(200, response)
  .onPut().reply(200, response)
  .onDelete().reply(200, response);

describe('Container: App', () => {
  let component;
  const buildComponent = () => {
    return shallow(<App classes={{}} />);
  };

  describe('Snapshots', () => {
    it('Should render correctly', () => {
      component = buildComponent();
      expect(shallowToJson(component)).toMatchSnapshot();
    });
  });

  describe('Methods', () => {
    it('Should add a place correctly', () => {
      component = buildComponent();
      component.instance().addPlace({ id: 1 });
      expect(component.state().places.length).toBe(1);
    });

    it('Should update a place correctly', () => {
      component = buildComponent();
      const newPlace = { id: 1, name: 'hola' };
      component.instance().updatePlace(newPlace);
      expect(component.state().places[0]).toBe(newPlace);
    });

    it('Should destroy a place correctly', () => {
      component = buildComponent();
      component.instance().removePlace(1);
      expect(component.state().places.length).toBe(0);
    });
  });

  describe('Actions', () => {
    it('Click on the help button should open the dialog', () => {
      component = buildComponent();
      component.find('#help').simulate('click');
      expect(component.state().help).toBeTruthy();
    });
  });
});