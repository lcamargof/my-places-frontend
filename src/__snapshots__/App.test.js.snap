// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: App Snapshots Should render correctly 1`] = `
<div
  className="App"
>
  <WithStyles(MainAppBar)
    onFilter={[Function]}
    places={Array []}
  />
  <Map
    onAdd={[Function]}
    onDelete={[Function]}
    onUpdate={[Function]}
    place={null}
    places={Array []}
  />
  <WithStyles(Button)
    aria-label="help"
    color="primary"
    id="help"
    onClick={[Function]}
    variant="fab"
  >
    <pure(HelpOutline) />
  </WithStyles(Button)>
  <HelpDialog
    onClose={[Function]}
    open={false}
  />
  <WithStyles(Component)
    onClose={[Function]}
    open={false}
    text=""
  />
</div>
`;
